web/modules/custom/joinup_core/joinup_core.post_update.php

```diff
@@ -48,3 +48,103 @@ function joinup_core_post_update_0106401(): void {
   $index->reindex();
   $index->indexItems(-1, 'entity:taxonomy_term');
 }
+
+/**
+ * Remove path alias duplicates.
+ */
+function joinup_core_post_update_0106402(?array &$sandbox = NULL): string {
+  $db = \Drupal::database();
+  if (!isset($sandbox['duplicate_pids'])) {
+    // Get all duplicate path alias IDs.
+    $sandbox['duplicate_pids'] = $db->query("SELECT p.id
+    FROM {path_alias} p
+    LEFT JOIN (
+      -- This sub-query returns all alias duplicates of English aliases.
+      SELECT
+        MAX(id) AS valid_id,
+        COUNT(*) AS duplicates_count,
+        path
+      FROM {path_alias}
+      WHERE langcode = 'en'
+      GROUP BY path
+      HAVING duplicates_count > 1
+    ) valid_aliases ON p.path = valid_aliases.path
+    WHERE valid_aliases.valid_id IS NOT NULL
+    AND p.id <> valid_aliases.valid_id
+    -- Only select English aliases.
+    AND p.langcode = 'en'")->fetchCol();
+    $sandbox['progress'] = 0;
+    $sandbox['total'] = count($sandbox['duplicate_pids']);
+  }
+
+  $to_delete = array_splice($sandbox['duplicate_pids'], 0, 1000);
+  $db->delete('path_alias_revision')
+    ->condition('id', $to_delete, 'IN')
+    ->execute();
+  $db->delete('path_alias')
+    ->condition('id', $to_delete, 'IN')
+    ->execute();
+  $sandbox['progress'] += count($to_delete);
+
+  if ($sandbox['#finished'] = (int) empty($sandbox['duplicate_pids'])) {
+    \Drupal::entityTypeManager()->getStorage('path_alias')->resetCache();
+  }
+
+  // Also, remove the key-value pairs for entities that might have their paths
+  // set as persistent.
+  \Drupal::keyValue('pathauto_state.rdf_entity')->deleteAll();
+  \Drupal::keyValue('pathauto_state.node')->deleteAll();
+
+  return "Removed {$sandbox['progress']}/{$sandbox['total']}";
+}
+
+/**
+ * Update aliases for entities with the old alias.
+ */
+function joinup_core_post_update_0106403(?array &$sandbox = NULL): string {
+  $rdf_storage = \Drupal::entityTypeManager()->getStorage('rdf_entity');
+  $node_storage = \Drupal::entityTypeManager()->getStorage('node');
+  if (empty($sandbox['entity_ids'])) {
+    $sandbox['entity_ids']['rdf_entity'] = $rdf_storage->getQuery()->condition('rid', 'collection')->execute();
+    $sandbox['entity_ids']['rdf_entity'] += $rdf_storage->getQuery()->condition('rid', 'solution')->execute();
+    $sandbox['entity_ids']['rdf_entity'] += $rdf_storage->getQuery()->condition('rid', 'asset_release')->execute();
+    $sandbox['entity_ids']['rdf_entity'] += $rdf_storage->getQuery()->condition('rid', [
+      'collection',
+      'solution',
+      'asset_release',
+    ], 'NOT IN')->execute();
+    $sandbox['entity_ids']['node'] = [];
```

Comment:
Oops! After running the post update I'm doing:

```sql
SELECT * FROM path_alias WHERE alias NOT LIKE '/collection/%' AND path LIKE '/node/%'
```

...and I'm still getting ~6.600 records.

Reply:
:( When you forget to revert test changes..... :(